spec/mauth_client_spec.rb

```diff
@@ -52,231 +66,520 @@ def (::Rails).logger
         mc = MAuth::Client.new(config)
         # can't directly compare the OpenSSL::PKey::RSA instances
         expect(key.class).to eq(mc.private_key.class)
         expect(key.to_s).to eq(mc.private_key.to_s)
       end
     end
+
+    it 'correctly initializes with authenticate_with_only_v2 as true with boolean true or string "true"' do
+      [true, 'true'].each do |authenticate_with_only_v2|
+        [{:authenticate_with_only_v2 => authenticate_with_only_v2}, {'authenticate_with_only_v2' => authenticate_with_only_v2}].each do |config|
+          mc = MAuth::Client.new(config)
+          expect(mc.authenticate_with_only_v2).to eq(true)
+        end
+      end
+    end
+
+    it 'correctly initializes with authenticate_with_only_v2 as false with any other values' do
+      ['tru', false, 'false', 1, 0, nil, ''].each do |authenticate_with_only_v2|
+        [{:authenticate_with_only_v2 => authenticate_with_only_v2}, {'authenticate_with_only_v2' => authenticate_with_only_v2}].each do |config|
+          mc = MAuth::Client.new(config)
+          expect(mc.authenticate_with_only_v2).to eq(false)
+        end
+      end
+    end
+
+    it 'correctly initializes with sign_requests_with_only_v2 as true with boolean true or string "true"' do
+      [true, 'true'].each do |sign_requests_with_only_v2|
+        [{:sign_requests_with_only_v2 => sign_requests_with_only_v2}, {'sign_requests_with_only_v2' => sign_requests_with_only_v2}].each do |config|
+          mc = MAuth::Client.new(config)
+          expect(mc.sign_requests_with_only_v2).to eq(true)
+        end
+      end
+    end
+
+    it 'correctly initializes with sign_requests_with_only_v2 as false with any other values' do
+      ['tru', false, 'false', 1, 0, nil].each do |sign_requests_with_only_v2|
+        [{:sign_requests_with_only_v2 => sign_requests_with_only_v2}, {'sign_requests_with_only_v2' => sign_requests_with_only_v2}].each do |config|
+          mc = MAuth::Client.new(config)
+          expect(mc.sign_requests_with_only_v2).to eq(false)
+        end
+      end
+    end
   end
 
   require 'mauth/request_and_response'
   class TestSignableRequest < MAuth::Request
     include MAuth::Signed
     attr_accessor :headers
+
     def merge_headers(headers)
       self.class.new(@attributes_for_signing).tap{|r| r.headers = (@headers || {}).merge(headers) }
     end
+
     def x_mws_time
       headers['X-MWS-Time']
     end
+
     def x_mws_authentication
       headers['X-MWS-Authentication']
     end
+
+    def mcc_authentication
+      headers['MCC-Authentication']
+    end
+
+    def mcc_time
+      headers['MCC-Time']
+    end
   end
 
   describe '#signed' do
-    before { @request = TestSignableRequest.new(:verb => 'PUT', :request_url => '/', :body => 'himom') }
+    it 'adds only X-MWS-Time and X-MWS-Authentication headers when signing with v1 override' do
+      expect(v1_signed_req.headers.keys).to include('X-MWS-Authentication')
```

Comment:
I believe these statements can be combined, per https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/include-matcher:
`expect(v1_signed_req.headers.keys).to include('X-MWS-Authentication', 'X-MWS-Time')`
Edit: Maybe these changes just being copied over? 😅 Bit tricky to review the spec file, maybe Jordi's comment about splitting into separate files would help here too

Reply:
yeah, then at least the new files can be rubocopped